Draft: Emoji to represent GTFS route type

exxonvaldez opened the pull request:
<!-- Does this PR fix any issues? If so, uncomment the below and put in the right number(s).
     You need to have a separate "Fixes #xyz" sentence for each issue you fix.
     Of course, erase issue numbers you don't need.
-->

This a partial prototype for #457, show route type in UI, to get early feedback and to hand over if anybody else is interested in taking over.


<!-- Delete any of these headings if they aren't needed or applicable -->

## Proposed changes

Using native (UTF-8) emojis, wrapped in a span for accessibility.  Created constants to represent emoji symbol and label by GTFS route type.

Added emoji to route names in the route table, and in the bread crumb, just to see how it looks.  Alternate emojis are noted in the source code.

## Screenshot

Route table:

![Screen Shot 2019-12-02 at 11 15 31 PM](https://user-images.githubusercontent.com/44861283/70028514-b1f33480-1559-11ea-9de3-202011bf0904.png)

Stop to stop:

![Screen Shot 2019-12-02 at 11 16 29 PM](https://user-images.githubusercontent.com/44861283/70028562-cdf6d600-1559-11ea-91fe-1610195ee7c1.png)

(Could add emojis to route picker, "from stop" dropdown label, and map, depending on how far we want to take it).

## Link to demo, if any

n/a

hathix commented:
Looks cool! Emojis add a pop of color and fun, which I think the app really benefits from. I like the idea of adding the emoji to the map, though then we'd probably want to replace all instances of that generic train icon with an emoji... which might get too busy. So I think your current approach is a good balance

youngj commented:
Emojis are different on each OS (e.g. https://emojipedia.org/bus/), so the UI would be more consistent if we used our own icons. Also I think the colors are too distracting especially when repeated for each route in the table. Single color icons would probably look cleaner and more professional.

exxonvaldez commented:
Here's some options for implementing emoji or icons:

- Native emoji is the easiest to implement but varies by client platform.

- This is an emoji picker component that seems to have the emoji images for 'apple', 'google', 'twitter', 'emojione', 'messenger', 'facebook' if we preferred the emoji for a specific platform (not sure if this is legal redistribution/reuse):

https://github.com/missive/emoji-mart

- We can search through the Material UI icons, these are already monochrome and available in several styles per icon (solid, outlined, etc.):

https://material-ui.com/components/material-icons/

- We can also look on Google for free icons:

https://www.google.com/search?q=bus+train+icons+free

I think the next best/easiest iteration to try would be Material UI icons in the route table (and if successful, then see how they look in the breadcrumbs and map).

hathix commented:
[MUI's icons](https://material-ui.com/components/material-icons/) are a good option too. A quick rundown of which icons it supports, for the record:

Tram: yes, `tram`
Subway: yes, `subway`
Rail: yes, `train`
Bus: yes, `directionsbus`
Ferry: yes, `directionsboat`
Cable car:  no
Gondola: no
Funicular: no

I think this is emblematic of the broader problem of using anything besides emojis: it'll be hard to get icons for that long tail of transport modes.

Using random icons on Google might lead to a mishmash of different styles, and the images may not be high quality.

Another option is Font Awesome, but it also lacks cable cars, gondolas, and funiculars.

We could also use Mozilla's or OpenMoji's emoji pictures ([example](https://emojipedia.org/aerial-tramway/)), which would probably be open source.

hathix commented:
I would say that for a "minimum viable product" it's best to go with the good-enough, platform-specific emojis. We can add a fancier alternative later.

exxonvaldez commented:
Re: open source emojis, good point.  A few more emoji sets that are open:  Google and Twitter (https://blog.emojipedia.org/who-owns-emoji/ )

hathix commented:
The Google one is more like an international font than an emoji pack. [Twemoji](https://twemoji.twitter.com/) is super nice though, and very recognizable. And it offers SVGs. We can use that.

hathix commented:
Oh man, you can programmatically create Twemoji! That's a great feature.

zachzwy commented:
I like @youngj 's idea of single color emoji, considering there are already a few different colors on the current route table. We don't want the route's emoji color to mis-indicate its quality.

Maybe all black icon?
<img width="717" alt="Screen Shot 2019-12-10 at 9 35 56 PM" src="https://user-images.githubusercontent.com/22914663/70594343-171bdb00-1b95-11ea-95d9-76fff7c5230a.png">

exxonvaldez commented:
I've updated this PR with the emojis commented out and added the Material UI icons @hathix suggested.  The solid icons were too dark, so I switched some to outlined.  Either way, the tram and bus icons are very similar, so the result is not compelling.  It was pretty easy to try out, at least.

![Screen Shot 2019-12-11 at 1 19 56 PM](https://user-images.githubusercontent.com/44861283/70661643-905d1180-1c19-11ea-83ae-519de290b5dc.png)

![Screen Shot 2019-12-11 at 1 24 22 PM](https://user-images.githubusercontent.com/44861283/70661654-96eb8900-1c19-11ea-8a35-aaa380383e45.png)

hathix commented:
The UI needs some polishing, says Terence. Not ready to merge this. We'd like to get some ui people to look at this

exxonvaldez commented:
Closing this out.  This PR can still be found easily via the open issue #457.
